src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -0,0 +1,217 @@
+            AccountKey = accountKey;
+            ContainerName = containerName;
+            log = loggingHelper;
+        }
+
+        public string FeedUrl
+        {
+            get
+            {
+                return AzureHelper.GetContainerRestUrl(AccountName, ContainerName);
+            }
+        }
+
+        public async Task<bool> CheckIfFeedExists()
+        {
+            string url = $"{FeedUrl}?restype=container";
+            using (HttpClient client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Clear();
+                var request = AzureHelper.RequestMessage("GET", url, AccountName, AccountKey).Invoke();
+                using (HttpResponseMessage response = await client.SendAsync(request))
+                {
+                    if (response.IsSuccessStatusCode)
+                    {
+                        log.LogMessage(
+                            MessageImportance.Low,
+                            $"Container {ContainerName} exists for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                    }
+                    else
+                    {
+                        log.LogMessage(
+                            MessageImportance.Low,
+                            $"Container {ContainerName} does not exist for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                    }
+                    return response.IsSuccessStatusCode;
+                }
+            }
+        }
+
+        public string GenerateIndexes(IEnumerable<string> items, string relativePath)
+        {
+            log.LogMessage(MessageImportance.Low, $"START generating indexes for {relativePath}");
+            GenerateRootServiceIndex(AccountName, relativePath);
+            foreach (var package in items)
+            {
+                Tuple<string, NuGetVersion> blobPath = CalculateBlobPath(package);
+                GeneratePackageServiceIndex(AccountName, Path.Combine(relativePath, blobPath.Item1), blobPath.Item2.ToFullString());
+            }
+            log.LogMessage(MessageImportance.Low, $"DONE generating indexes for {relativePath}");
+            return Path.Combine(Directory.GetCurrentDirectory(), "tmp");
+        }
+
+        public Tuple<string, NuGetVersion> CalculateBlobPath(string package)
+        {
+            using (var reader = new PackageArchiveReader(package))
+            using (var nuspecStream = reader.GetNuspec())
+            {
+                NuspecReader nuspec = new NuspecReader(nuspecStream);
+                return new Tuple<string, NuGetVersion>(nuspec.GetId(), nuspec.GetVersion());
+            }
+        }
+
+        private void GeneratePackageServiceIndex(string accountName, string redirectUrl, string value)
+        {
+            string pathToTempFolder = Path.Combine(Directory.GetCurrentDirectory(), "tmp", redirectUrl);
```

Comment:
Using the tmp directory a lot causes a bunch of garbage. What is the strategy for cleaning up the garbage?

Reply:
We could add a flag that deletes after publish in this task, else we can have the automated cleanup in official builds handle this.

Reply:
You are currently putting that tmp directory under the current working directory which may or may not ever be cleaned up. I think it is probably best to have someone pass you in a directory to work with or to use the real temp directory.